AssertionError when rewriting open(path, u"r")

Using pyupgrade 2.19.1 + Python 3.8.2 + macOS Big Sur 11.4, I get an AssertionError thrown for this input:
```py
text = open("file.txt", u"r").read()
```
 
To reproduce, put that single line in a `test.py` and run `pyupgrade --py36-plus test.py`. The output:
```
Traceback (most recent call last):
  File "/Users/foolip/Library/Python/3.8/bin/pyupgrade", line 10, in <module>
    sys.exit(main())
  File "/Users/foolip/Library/Python/3.8/lib/python/site-packages/pyupgrade/_main.py", line 928, in main
    ret |= _fix_file(filename, args)
  File "/Users/foolip/Library/Python/3.8/lib/python/site-packages/pyupgrade/_main.py", line 863, in _fix_file
    contents_text = _fix_plugins(
  File "/Users/foolip/Library/Python/3.8/lib/python/site-packages/pyupgrade/_main.py", line 123, in _fix_plugins
    callback(i, tokens)
  File "/Users/foolip/Library/Python/3.8/lib/python/site-packages/pyupgrade/_plugins/open_mode.py", line 47, in _fix_open_mode
    raise AssertionError(f'unreachable: {mode!r}')
AssertionError: unreachable: ' u"r"'
```

That's this assert being hit:
https://github.com/asottile/pyupgrade/blob/e1485c79860827b7880f8c5a40fc8ef02456fcc6/pyupgrade/_plugins/open_mode.py#L47

The assertion is *not* hit if the input is change to this, dropping the `u` string prefix:
```py
text = open("file.txt", "r").read()
```

I haven't looked closer than this.

Original code: https://github.com/web-platform-tests/wpt/blob/0f986ae5b0865c0bef78e41e35d144e30a14b2e7/eventsource/resources/cors.py

--- a/pyupgrade/_plugins/open_mode.py
+++ b/pyupgrade/_plugins/open_mode.py
@@ -34,7 +34,7 @@ def _fix_open_mode(i: int, tokens: List[Token], *, arg_idx: int) -> None:
     func_args, end = parse_call_args(tokens, j)
     mode = tokens_to_src(tokens[slice(*func_args[arg_idx])])
     mode_stripped = mode.split('=')[-1]
-    mode_stripped = mode_stripped.strip().strip('"\'')
+    mode_stripped = ast.literal_eval(mode_stripped.strip())
     if mode_stripped in U_MODE_REMOVE:
         delete_argument(arg_idx, tokens, func_args)
     elif mode_stripped in U_MODE_REPLACE_R:

--- a/tests/features/open_mode_test.py
+++ b/tests/features/open_mode_test.py
@@ -62,6 +62,7 @@ def test_fix_open_mode_noop(s):
             'open(mode="r", encoding="UTF-8", file="t.py")',
             'open(encoding="UTF-8", file="t.py")',
         ),
+        pytest.param('open(f, u"r")', 'open(f)', id='string with u flag'),
     ),
 )
 def test_fix_open_mode(s, expected):

